students/EricAdams/session09/circle.py

```diff
@@ -0,0 +1,38 @@
+#! /usr/bin/env python
+# circle.py
+# A Circle can be defined by either specifying the radius or the diameter,
+# and the user can query the circle for either its radius or diameter.
+
+# Other abilities of a Circle instance:
+
+# Compute the circle’s area
+# Print the circle and get something nice
+# Be able to add two circles together
+# Be able to compare two circles to see which is bigger
+# Be able to compare to see if there are equal
+# (follows from above) be able to put them in a list and sort them
+
+import math
+
+
+class Circle():
+    def __init__(self, radius):
+        self.radius = radius
+    _diameter = 0
```

Comment:
You really don’t want to be storing the diameter. You are re-calculating it whenever you get it in any case.

And if you Store it, it could get out of sync with radius...